pkg/kubectl/cmd/annotate.go

```diff
@@ -193,7 +193,11 @@ func (o AnnotateOptions) RunAnnotate() error {
 			return err
 		}
 
-		name, namespace, obj := info.Name, info.Namespace, info.Object
+		obj, err := resource.AsVersionedObject([]*resource.Info{info}, false, info.Mapping.GroupVersionKind.Version)
```

Comment:
also, looks like AsVersionedObject attempts to use the version you give it, then falls back to `info.Mapping.GroupVersionKind.GroupVersion().String()`, so I guess (?) we can ask it to convert to the version we've negotiated with the server, and if it can't it'll convert to the versioned mapping?